tree_builder.py

```diff
@@ -116,6 +122,22 @@ def build_discussion_dict(lines):
 
     return discussion_tree
 
+def front_augmentation(args):
```

Comment:
A comment explaining the purpose of this method (the definition of 'augmentation' in this context) might help future readability.